src/Jupyter/Visualization/JupyterSimulator.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Numerics;
+using Microsoft.Jupyter.Core;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+using Newtonsoft.Json;
+
+namespace Microsoft.Quantum.IQSharp.Jupyter
+{
+    public class JupyterSimulator : QuantumSimulator
+    {
+        private readonly IChannel Channel;
+        private readonly IConfigurationSource ConfigurationSource;
+        public JupyterSimulator(IChannel channel,
+                                IConfigurationSource configurationSource,
+                                bool throwOnReleasingQubitsNotInZeroState = true,
+                                uint? randomNumberGeneratorSeed = null,
+                                bool disableBorrowing = false)
+        : base(throwOnReleasingQubitsNotInZeroState, randomNumberGeneratorSeed, disableBorrowing)
+        {
+            Channel = channel;
+            ConfigurationSource = configurationSource;
+            // Default to setting the channel to be the handler for Message.
+            DisableLogToConsole();
+            OnLog += Channel.Stdout;
+        }
+
+        protected override QVoid Dump<T>(T target, IQArray<Qubit>? qubits = null)
```

Comment:
In order to make it a bit more flexible, what we can do is instead of extending the QuantumSimulator and override the `Dump` method, create two native implementations of `Dump` and `DumpMachine` (similar to [the ootb implementation](https://github.com/microsoft/qsharp-runtime/blob/4109c3bdd8f8048d1309b85bb81784ea61a2d7ff/src/Simulation/Simulators/QuantumSimulator/Dump.cs#L66)), and then instead of using a `JupyterSimulator`, call the `Register` method of the `QuantumSimulator` to override the default `DumpMachine` with this new implementation.
Doing it this way, would allow us to easily incorporate it on the Katas simulator (all it will need to do is do the same registration).
Let's talk tomorrow if you have questions...

Reply:
Good idea, I'll go on and do that, then.